compiler/src/test/kotlin/symboltest.kt

```diff
@@ -0,0 +1,145 @@
+package dk.aau.cs.d409f19
+
+import dk.aau.cs.d409f19.Utilities.Companion.compileProgram
+import dk.aau.cs.d409f19.Utilities.Companion.getParser
+import dk.aau.cs.d409f19.Utilities.Companion.getStateDecl
+import dk.aau.cs.d409f19.Utilities.Companion.getWorldDecl
+import dk.aau.cs.d409f19.cellumata.ErrorLogger
+import dk.aau.cs.d409f19.cellumata.ast.*
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.opentest4j.AssertionFailedError
+import java.lang.NullPointerException
+
+class SymbolTest {
+
+    /**
+     * Reset ErrorLogger as it is globally persistent and will interfere with tests
+     */
+    @BeforeEach
+    fun resetErrorLogger() {
+        ErrorLogger.reset()
+    }
+
+    /**
+     * Tests assignments statements which should be legal, given an identifier and value as parameter.
+     * If assertions fails, exceptions are caught, and false is returned to indicate a failed test to caller.
+     */
+    private fun <T> assignStmtPass(ident: String, value: T): Boolean {
+
+        val compilerData = compileProgram(getWorldDecl() + "\n\n" + getStateDecl(body = "let $ident = $value;"))
+
+        // Get SymbolTable for entire program
+        val symbolTable = compilerData.scopeChecker.getSymbolTable()
+        // Get SymbolTable for first subscope, which is first StateDecl
+        val stateSymbolTable = symbolTable.tables[0]
+
+        try {
+            // Get expression of AssignStmt which is value of the key of ident
+            val literal = (stateSymbolTable.symbols[ident] as AssignStmt).expr
+
+            // Assert identifier of variable is contained in first symbol scope
+            assertTrue(stateSymbolTable.symbols.containsKey(ident))
```

Comment:
Yes, good catch. Note that this NPE catch-block would find the issue and return false:
```kotlin
        } catch (e: NullPointerException) { // Catch NPE's which may be thrown due to looking up ident as key
            e.printStackTrace()
            return false
        }
```

I've moved the assertion above the lookup, such that it would fail with an AFE in the case that it did not exist. This allows for removing the NPE catch block and retaining the test. See commit: d0c5391